### PR TITLE
Add styles for disabled links and buttons [SANDBOX]

### PR DESCRIPTION
**Pull request by cwardzala**

### CODE REVIEW

**Review by nclayt0n — APPROVED** [review]

**Review by nsmedira — APPROVED** [review]
this stops the pointer events but doesn't stop keyboard events so one could still tab through buttons/links and follow the link

but probably an effective solution for most cases

**Comment by cwardzala** [comment]
@nsmedira good catch I will investigate a solution to fix that.

**Comment by cwardzala** [comment]
OK, I have landed on a solution using `useDisabledLinks`.

File: src/hooks/useDisabledLinks.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    useEffect(() => {
+        const preventDisabledAnchorClicks = (event) => {
+            const attrDisabled = event.target.getAttribute('disabled');
+            const tagName = event.target.tagName;
+            if (attrDisabled !== null && tagName === 'A') {
+                event.preventDefault();
+                return false;
+            }
+        };
+        window.document.addEventListener('click', preventDisabledAnchorClicks, true);
```
Comment (nsmedira) [inline src/hooks/useDisabledLinks.js:24]: good catch on setting `useCapture` argument to `true`

**Review by nsmedira — APPROVED** [review]
just implemented this with the "Submit" button I demonstrated in mp production flow that prompted this development—working excellently!

**Comment by nsmedira** [comment]
i just noticed that `pointer-events: none` prevents the hover state. 

what kind of behavior do we want when hovering disabled buttons or links? if we wanted to allow a hover state, we could remove the changes to `_buttons.scss` and in `_links.scss` replace `pointer-events: none` with `cursor: default` so that disabled links and buttons have equivalent hover behavior

**Comment by cwardzala** [comment]
@nsmedira I would say that there should be no hover state when disabled. This PR adds `pointer-events: none;` for both `a` and `.btn` styles.

**Pull request merged**